plugins/PrivacyManager/tracker.js

```diff
@@ -0,0 +1,35 @@
+/**
+ * Matomo - free/libre analytics platform
+ *
+ * @link https://matomo.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ *
+ */
+
+(function () {
+
+    function init() {
+        if ('object' === typeof window && !window.Matomo) {
+            // Matomo is not defined yet
+            return;
+        }
+
+        // disable cookies and remove them when a tracker is created
+        window.Matomo.on('TrackerSetup', function(tracker) {
+            tracker.setCookieConsentGiven=function(){};
+            tracker.rememberCookieConsentGiven=function(){};
+            tracker.disableCookies();
+        });
+    }
+
+    if ('object' === typeof window.Matomo) {
+        init();
```

Comment:
Note: This feature will only work correctly if the `matomo.js / piwik.js` is writable. It won't 100% work when it is loaded separately and I reckon people would probably also not load it as a separate file and we would then rather rely on recommending people to call `disableCookies` and ignoring any cookie server side.

To be safe we could iterate over all existing `Matomo.getAsyncTrackers()` and call the setup method as well for each tracker. However, it would not work for any tracker created using `getTracker()` plus at this point it might be too late for any async tracker as they might have created cookies already. That be not a big issue though, we'll make users aware of this in the setting maybe.

Reply:
Sure. That also won't work when cookie less tracking is "forced" for a single site only. So we need to advice people to call `disableCookies` nevertheless.
Actually I was thinking about having a separate piwik_no_cookies.js that maybe even does not contain any of the code that uses cookies at all. That would make it easy to ensure no cookies would be set... But would be of course cause more work to maintain both files